### PR TITLE
fix #1830. credit from https://github.com/danielgindi/Charts/pull/2049

### LINKED ISSUE #1830: A problem of CombinedChartView's YAxis
I think there might be a problem with the YAxis of CombinedChartView.

I have a CombinedChartView with one bar and one line. 
The bar depends on the left YAxis.
The line depends on the right YAxis.
After the dependencies are well configured, I get the chart below, which is definitely not what I want.
The labels of the left YAxis and the right YAxis are the same even though y values are totally different.
I'm not sure if it's designed like that intentionally. If so, just ignore what I say below.

simulator screen shot 2016 11 14 8 56 22

I am confused, so I review the source code. Finally I find a problem of the function calcMinMax() in CombinedChartData.swift.

            if data.yMax > _leftAxisMax
            {
                _leftAxisMax = data.yMax
            }

            if data.yMin < _leftAxisMin
            {
                _leftAxisMin = data.yMin
            }

            if data.yMax > _rightAxisMax
            {
                _rightAxisMax = data.yMax
            }

            if data.yMin < _rightAxisMin
            {
                _rightAxisMin = data.yMin
            }

It seems that dataset's ydependency is NOT considered at all. So I change it.

            for i in 0 ..< sets.count
            {
                let dataset = sets[i]
                if dataset.axisDependency == .left
                {
                    if dataset.yMax > _leftAxisMax
                    {
                        _leftAxisMax = dataset.yMax
                    }
                    if dataset.yMin < _leftAxisMin
                    {
                        _leftAxisMin = dataset.yMin
                    }
                } else { // right
                    if dataset.yMax > _rightAxisMax
                    {
                        _rightAxisMax = dataset.yMax
                    }
                    if dataset.yMin < _rightAxisMin
                    {
                        _rightAxisMin = dataset.yMin
                    }
                }
            }

Now I get what I want.

simulator screen shot 2016 11 14 8 55 02

I think it might be helpful to someone like myself.

### PR DESCRIPTION
**Pull request by liuxuan30**
fix #1830. credit from https://github.com/danielgindi/Charts/pull/2049

### CODE REVIEW

**Comment by liuxuan30** [comment]
@danielgindi @petester42 mind to take a look at this? especially the test portion. The fix should be positive.

**Comment by danielgindi** [comment]
I'll look at it :-)

**Comment by liuxuan30** [comment]
@danielgindi have you got some time for this? Planning to merge

File: Source/Charts/Data/Implementations/Standard/CombinedChartData.swift [hunk 1]
```diff
@@ -136,25 +136,32 @@ open class CombinedChartData: BarLineScatterCandleBubbleChartData
...
-            }
-            
-            if data.yMax > _rightAxisMax
-            {
-                _rightAxisMax = data.yMax
-            }
-            
-            if data.yMin < _rightAxisMin
+
+            for i in 0 ..< sets.count
```
Comment (jjatie) [inline Source/Charts/Data/Implementations/Standard/CombinedChartData.swift]: `for dataset in sets`

**Comment by liuxuan30** [comment]
@jjatie I will find a time to fix the conflict for this PR. It includes combined chart unit test, which I think should be useful before merging your swift PRs.

File: Source/Charts/Data/Implementations/Standard/CombinedChartData.swift (hunk 1 above)
Comment (liuxuan30) [reply Source/Charts/Data/Implementations/Standard/CombinedChartData.swift]: done

**Pull request merged**